### PR TITLE
ENH: allow passing freq=None to DatetimeIndex/TimedeltaIndex

### PR DESCRIPTION
**Pull request by jbrockmendel**
cc @jreback discussed elsewhere

### CODE REVIEW

File: pandas/core/arrays/datetimelike.py [hunk 1]
```diff
@@ -1674,7 +1674,7 @@ def validate_endpoints(closed):
     return left_closed, right_closed
 
 
-def validate_inferred_freq(freq, inferred_freq, freq_infer):
+def validate_inferred_freq(freq, inferred_freq, freq_infer, freq_inherit):
```
Comment (jreback) [inline pandas/core/arrays/datetimelike.py:1677]: this is pretty confusing, how about just allowing freq_infer to be True/False/None.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/arrays/datetimelike.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/arrays/datetimelike.py:1677]: so freq_infer=None would mean inherit but dont do active inference?  im not sure thats clearer

Comment (jbrockmendel) [reply pandas/core/arrays/datetimelike.py]: updated with a simpler implementation

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
looks good. if you can merge master and ping on green.

**Comment by jbrockmendel** [comment]
ping

**Pull request merged**